pkg/arrai/sysl.arrai

```diff
@@ -0,0 +1,43 @@
+# Performs transformations of Sysl models, producing "better" Sysl models.
+
+# Invokes a macro on a string as if it were source code at parsing time.
+let invokeMacro = \macro \s
+    macro -> (//dict(.@transform) >>> \rule \fn fn(//grammar.parse(.@grammar, rule, s))).@value;
```

Comment:
you can iterate through tuples with the `:>` operator. For example:

```
@> (a: 1, b: 2, c: 3) :> . + 1
(a: 2, b: 3, c: 4)
```

Reply:
Indeed, but I need the name of the tuple attribute in order to compute its new value, not just the current value.